tests/commons.py

```diff
@@ -114,25 +115,35 @@ def event_loop():
         loop.close()
 
 
-async def create_user_with_groups(groups: list[GroupType]) -> models_core.CoreUser:
+async def create_user_with_groups(
+    groups: list[GroupType],
+    user_id: str | None = None,
+    email: str | None = None,
+    password: str | None = None,
+    name: str | None = None,
+    firstname: str | None = None,
+    floor: FloorsType = FloorsType.Autre,
+) -> models_core.CoreUser:
     """
     Add a dummy user to the database
-    The user will be named with its user_id. Email and password will both be its user_id
+    If a core_user is not provided, the user will be named with its user_id and email and password will both be its user_id
 
     The user will be added to provided `groups`
     """
-    user_id = str(uuid.uuid4())
 
-    password_hash = security.get_password_hash(user_id)
+    user_id = user_id or str(uuid.uuid4())
+    assert user_id is not None
+    password_hash = security.get_password_hash(password or get_random_string())
```

Comment:
The objective is to create an user with random data. We can pass a password to this function if we need to use a specific value, but we want the user to be randomly generated otherwise.